### ISSUE TITLE
Incorrect match result when in PHP

### ISSUE THREAD

**Issue by zhanglianxin** [firasdib/Regex101#1255]
## Bug Description

regex: `([^a-zA-Z0-9])([a-zA-Z0-9]{11,})(?:\1)`
test string: `￥IyB1YBGsOT5￥`
match information as below: 
```
Full match | 0-13 | ￥IyB1YBGsOT5￥
Group 1. | 0-1 | ￥
Group 2. | 1-12 | IyB1YBGsOT5
```

![](https://imgur.com/52asbKa.png)

### Reproduction steps

<https://regex101.com/r/Ld1ZAs/1/>

### Expected Outcome

```php
<?php

$regex = '/([^a-zA-Z0-9])([a-zA-Z0-9]{11,})(?:\1)/';
$str = '￥IyB1YBGsOT5￥';
preg_match_all($regex, $str, $matches);
var_dump($matches);
/*
array(3) {
  [0]=>
  array(0) {
  }
  [1]=>
  array(0) {
  }
  [2]=>
  array(0) {
  }
}
*/
```

![](https://i.imgur.com/mdYtQ6T.png)

Thanks for your great work.

**Comment by Doqnach** [comment]
This looks correct? Group 0 is full match, and like regex101, there are two match groups in the regex.
Not sure why your PHP output shows empty arrays though...

If you use the `u` flag in PHP it does show correctly: https://ideone.com/OeuEl8

**Comment by zhanglianxin** [comment]
> This looks correct? Group 0 is full match, and like regex101, there are two match groups in the regex.
> Not sure why your PHP output shows empty arrays though...
> 
> If you use the `u` flag in PHP it does show correctly: https://ideone.com/OeuEl8

Yeah, it works, thanks buddy. Regex101 may need some changes.

**Comment by Doqnach** [comment]
@zhanglianxin what do you feel needs changing?

**Comment by zhanglianxin** [comment]
> @zhanglianxin what do you feel needs changing?

Change its default output when without the `u` flag.

**Comment by Doqnach** [comment]
I have no idea why php gives this behaviour... but to me to seems like PHP is at fault here (not being able to handle an UTF-8 string if the `u` flag is not provided), and the regex101 website behaves as expected.

**Comment by zhanglianxin** [comment]
> I have no idea why php gives this behaviour... but to me to seems like PHP is at fault here (not being able to handle an UTF-8 string if the `u` flag is not provided), and the regex101 website behaves as expected.

**TL;DR**: You're right, the regex101 is right.

Well, I found my PHP (Version 7.4.0 on macOS installed via Homebrew) uses the `pcre` not the `pcre2`. 

`brew uses --install | grep php`, php is there. And `php --re pcre`, here is the result.

![](https://imgur.com/TmrzKv8.png)

As the pcre [man page][1] says:
> The  current implementation of PCRE corresponds approximately with Perl
> 5.12, including support for UTF-8/16/32  encoded  strings  and  Unicode
> general  category  properties. However, **UTF-8/16/32 and Unicode support
> has to be explicitly enabled; it is not the default**. The Unicode tables
> correspond to Unicode release 6.3.0.

and [the page][2] says:
> A special option is the `/u` which turns on the Unicode matching mode, instead of the default 8-bit matching mode. You should specify `/u` for regular expressions that use `\x{FFFF}`, `\X` or `\p{L}` to match Unicode characters, graphemes, properties or scripts. PHP will interpret `'/regex/u'` as a UTF-8 string rather than as an ASCII string.

@Doqnach thanks for your help again.

[1]: https://www.pcre.org/pcre.txt
[2]: https://www.regular-expressions.info/php.html
